lib/ae_mdw/contract.ex

```diff
@@ -346,6 +348,26 @@ defmodule AeMdw.Contract do
     Enum.group_by(get_events(micro_block), fn {_event_name, %{tx_hash: tx_hash}} -> tx_hash end)
   end
 
+  @spec maybe_resolve_contract_pk(name_pubkey() | contract_pubkey()) :: contract_pubkey()
+  def maybe_resolve_contract_pk(contract_or_name_pk) do
+    contract_or_name_pk
+    |> case do
+      <<217, 52, 92, _rest::binary>> = name_pk ->
+        "contract_pubkey"
+        |> :aec_chain.resolve_namehash(name_pk)
+        |> case do
+          {:ok, {:id, :contract, pubkey}} ->
```

Comment:
Can we avoid exposing the aeser_id structure and instead use the getters to retrieve the value? Something like
```elixir
{:ok, contract_id} -> :aeser_id.specialize(contract_id, :contract)
```